tests/Add-DbaAgDatabase.Tests.ps1

```diff
@@ -38,49 +47,130 @@ Describe "Add-DbaAgDatabase" -Tag "UnitTests" {
     }
 }
 
-Describe "Add-DbaAgDatabase" -Tag "IntegrationTests" {
+Describe "Add-DbaAgDatabase IntegrationTests" -Tag "IntegrationTests" {
     BeforeAll {
-        $null = Get-DbaProcess -SqlInstance $TestConfig.instance3 -Program 'dbatools PowerShell module - dbatools.io' | Stop-DbaProcess -WarningAction SilentlyContinue
-        $server = Connect-DbaInstance -SqlInstance $TestConfig.instance3
-        $agname = "dbatoolsci_addagdb_agroup"
-        $dbname = "dbatoolsci_addagdb_agroupdb"
-        $newdbname = "dbatoolsci_addag_agroupdb_2"
-        $server.Query("create database $dbname")
-        $backup = Get-DbaDatabase -SqlInstance $TestConfig.instance3 -Database $dbname | Backup-DbaDatabase
+        # We want to run all commands in the BeforeAll block with EnableException to ensure that the test fails if the setup fails.
+        $PSDefaultParameterValues['*-Dba*:EnableException'] = $true
+
+        # We remove all the calls to Get-DbaProcess and Stop-DbaProcess as they are not needed with correct tests.
+        #$null = Get-DbaProcess -SqlInstance $TestConfig.instance3 -Program 'dbatools PowerShell module - dbatools.io' | Stop-DbaProcess -WarningAction SilentlyContinue
```

Comment:
can be a "suggestion" but our tests are pretty much a bad bunch that needs to be tamed. getting and stopping shouldn't be the standard but also acceptable.

Reply:
It's mostly for the situation where someone writes a new test. That test should not include this line but work without. And I would like to remove these line to find those tests, that couse those problems. I already found some - and I like to find the others as well.